### DISCUSSION TITLE
Unable to view or use Page Title in search results for a Site Page using Microsoft Search as the Data Source.

### DISCUSSION THREAD

**Discussion by aspiringcodehead** [microsoft-search/pnp-modern-search#3884] [General]
Hello,

I'm not sure if this is a bug or not so I thought I should start here.

I have been trying to use the PnP Modern Search webparts to create a new search page for our Intranet, using Microsoft search as the data source. While I have been successful in creating a decent looking layout for our documents, I can't seem to get the Title of Site Pages to show up in the layout. I don't even see a Title or Name in the JSON results while looking at the debug layout, which makes me wonder if this is a bug.

This is also working fine for me when I use SharePoint as the data source, but I'm being asked to leverage Microsoft search instead. I've tried mapping a Refinable String to the Title field but that didn't work either, as well as other misc. things I've tried. Is there a way to retrieve and leverage the Title of a Site Page so I can show the title in the layout/search results, while using MS Search as the data source? What am I missing?

Thanks in advance for your time!

I have attached the JSON 
[pnp ms search debug site page json.txt](https://github.com/user-attachments/files/16447346/pnp.ms.search.debug.site.page.json.txt)
from the debug view for the page (sensitive info replaced with ~~redacted~~).

**Comment by kasperbolarsen** [comment]
I really haven't had the time to play with Microsoft search for content in SharePoint , as SharePoint Search seems to do the job with out the hassle. 
However,  in your case you might be able to get the title if you map a RefinableString as per here https://www.m365thinking.com/post/fix-one-of-the-problems-in-sharepoint-search-make-title-sortable

Please let us know if it solved the issue

**Reply by aspiringcodehead** [reply]
Thank you for your response and suggestion! Unfortunately I haven't been able to get the PnP Search results with MS search to return the mapped RefinableString for SitePages, despite adding it to the Selected Fields in the webpart. I do see the mapped RefinableString returned for documents however.

Trying to leverage Microsoft search has definitely been a hassle. We recently migrated from Classic SharePoint to Modern and I was asked to implement MS Search for our Intranet as my colleague and supervisor had concerns that we might be missing out on features and/or functionality if we stuck with SharePoint search. I'm left with the impression that some aspects of MS Search maybe aren't quite ready for production use (at least for us). Would you tend to agree? Is there any current advantage to using MS search over SharePoint search or vice-versa? At this point, I am planning to proceed with using the PnP webparts with SharePoint search (and revisit MS Search at a later date), but I still need to make the case to my colleague(s) and supervisor.

It seems like the Title field should be standard for a Site Page yet it isn't returned by default from the Microsoft search results within the PnP Search results webpart, using the debug layout. This issue appears to be for Site Pages only, as Documents do have the Title field returned. Should I submit this as a bug?

Thank you very much for your time!

**Reply by kasperbolarsen** [reply]
I can only agree with the opinion that MS Search isn't mature enough for us to switch from SharePoint Search. It seems like the team at MS responsible for search has been focused on Copilot for the last couple of years.
For instance, if you do a wildcard search for People in the Graph Explorer you will get 0 results. The reason is that the People search endpoint is using "?" as the wildcard where the other endpoints ( and the KQL spec specifies "*" as the wildcard ) ...... so ready for primetime? Nope
And on top of that, a People search in MS Search will not return all people as we are used to from SP Search, but only "my people", the bucket of people you are "related" to. 
I will keep an eye on MS Search and start publishing some guidance on how to transition to MS Search once it is ready, perhaps in 2-3 years time?

**Comment by aspiringcodehead** [comment]
Thank you very much for the response and info!
